tests/python/relay/test_df_pattern.py

```diff
@@ -0,0 +1,574 @@
+        if (tvm.ir.structural_equal(post.args[0], relay.const(0)) | 
+            tvm.ir.structural_equal(post.args[0], relay.const(0.0))):
+            return post.args[0]
+        else:
+            return post.args[1]
+
+    zero = (ExprPattern(relay.const(0)) | ExprPattern(relay.const(0.0)))
+    one = (ExprPattern(relay.const(1)) | ExprPattern(relay.const(1.0)))
+    add_pattern = wildcard() + zero
+    pattern_callbacks.append(DFPatternCallback(add_pattern, elwise_zero_callback))
+
+    sub_pattern = wildcard() - zero
+    pattern_callbacks.append(DFPatternCallback(sub_pattern, elwise_zero_callback))
+
+    mul_pattern = wildcard() * one
+    pattern_callbacks.append(DFPatternCallback(mul_pattern, elwise_one_callback))
+    
+    mul_zero_pattern = wildcard() * zero
+    pattern_callbacks.append(DFPatternCallback(mul_zero_pattern, return_zero_callback))
+
+    div_pattern = wildcard() / one
+    pattern_callbacks.append(DFPatternCallback(div_pattern, elwise_one_callback))
+
+    zero_div_pattern = zero / wildcard()
+    pattern_callbacks.append(DFPatternCallback(zero_div_pattern, return_zero_callback))
+
+    return rewrite(pattern_callbacks, expr);
+
+def test_algebraic_simplify():
+    x = relay.Var('x')
+    y = relay.Var('y')  
+
+    print(x + relay.const(0))
+    
+    one = relay.const(1)
+    zero = relay.const(0)
+    onef = relay.const(1.0)
+    zerof = relay.const(0.0)
+
+    assert algebraic_simplify(x + zero) == x
+    assert algebraic_simplify(x + zerof) == x
+    assert algebraic_simplify(zero + x) == x
+    assert algebraic_simplify(zerof + x) == x
+    
+    assert algebraic_simplify(x - zero) == x
+    assert algebraic_simplify(x - zerof) == x
+    
+    assert algebraic_simplify(x * one) == x
+    assert algebraic_simplify(x * onef) == x
+    assert algebraic_simplify(one * x) == x
+    assert algebraic_simplify(onef * x) == x
+    assert algebraic_simplify(x * zero) == zero
+    assert algebraic_simplify(x * zerof) == zerof
+    
+    assert algebraic_simplify(x / one) == x
+    assert algebraic_simplify(x / onef) == x
+    assert algebraic_simplify(zero / x) == zero
+    assert algebraic_simplify(zerof / x) == zerof
+
+    assert tvm.ir.structural_equal(algebraic_simplify((x + zero * y) / one + (y * one) - zero / x), x + y)
+
+if __name__ == "__main__":
+    #test_match_op()
+    #test_no_match_op()
+    #test_match_op_or()
+    #test_match_call()
+    #test_no_match_call()
+    #test_match_call_commutive()
+    #test_no_match_call_commutive()
+    #test_match_tuple()
+    #test_no_match_tuple()
+    #test_match_type()
+    #test_no_match_type()
+    #test_match_attr()
+    #test_no_match_attr()
+    #test_match_diamond()
+    #test_no_match_diamond()
+    #test_match_fake_diamond()
+    #test_rewrite()
+    #test_fuse_batchnorm()
+    #test_no_fuse_batchnorm()
+    #test_fuse_double_batchnorm()
+    #test_partial_fuse_double_batchnorm()
+    #test_fuse_batchnorm_commutation()
+    #test_match_dominator()
```

Comment:
enable these tests?

Reply:
They already run in CI via pytest, but I'll remove the comments so anyone running the tests manually doesn't hit issues. Thanks for the catch!